interfaces/MatlabToolbox/matlab_new/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1736 @@
+classdef ThermoPhase < handle
+    
+    properties
+        tpID
+        T % temperature
+        P % pressure
+        D % density
+        X % mole fractions
+        Y % mass fractions
+        H % enthalpy
+        S % entropy
+        U % internal energy
+        G % Gibbs free energye
+        basis
+    end
+
+    properties (Dependent)
+        V % specific volume
+        DP
+        DPX
+        DPY
+        HP
+        HPX
+        HPY
+        PV
+        PVX
+        PVY
+        SH
+        SHX
+        SHY
+        SP
+        SPX
+        SPY
+        ST
+        STX
+        STY
+        SV
+        SVX
+        SVY
+        TD
+        TDX
+        TDY
+        TH
+        THX
+        THY
+        TP
+        TPX
+        TPY
+        TV
+        TVX
+        TVY
+        UV
+        UVX
+        UVY
+        UP
+        UPX
+        UPY
+        VH
+        VHX
+        VHY
+    end
+    
+    methods
+        %% ThermoPhase class constructor
+
+        function tp = ThermoPhase(src, id)
+            % THERMOPHASE  ThermoPhase class constructor.
+            % t = ThermoPhase(src, id)
+            %
+            % :param src:
+            %     Input string of YAML, CTI, or XML file name.
+            % :param id:
+            %     ID of the phase to import as specified in the input file. (optional)
+            % :return:
+            %     Instance of class :mat:func:`ThermoPhase`
+            %
+            checklib;
+            if nargin > 2
+                error('ThermoPhase expects 1 or 2 input arguments.');
+            end
+            if nargin == 1
+                id = '-';
+            end
+            tp.tpID = calllib(ct, 'thermo_newFromFile', src, id);
```

Comment:
I notice that there is almost a complete lack of error handling in this interface. It is possible for most `clib` functions to return values indicating that a `CanteraError` was thrown internally (where the value indicating an error is different for a few different kinds of functions), and should be re-raised or otherwise reported to the user.

The need for this pattern of checking the error status after almost every call is one of the reasons that I strongly advocated for using code generation to implement this toolbox, because adding that error checking is going to be a _lot_ of repetitive boilerplate code.